### DISCUSSION TITLE
IsDirty set to true after load

### DISCUSSION THREAD

**Discussion by kcabral817** [MarimerLLC/csla#1668] [Questions] [answered]
I am having an awful time with the isDirty flag being set to true even when using BypassPropertyChecks and ((ICheckRules)obj).SuppressRuleChecking(); in the same method!

I'm using a factory class that uses Automapper to map from the data source to my BO.  
 var obj = ObjectFactory.Create<EquipmentItemInfoPersist>(); 

            using (BypassPropertyChecks(obj))
            {
                MarkOld(obj);
                ((ICheckRules)obj).SuppressRuleChecking();                 
                MapContainer.Map(info, obj);
                MarkOld(obj);
             }

Here is the status at the MarkOld(obj) line:
  
		IsBusy			false	bool
		IsBypassingPropertyChecks	true	bool
		IsChild			false	bool
		IsDeleted               false	bool
		IsDirty			true	bool
		IsNew			false	bool
		IsSavable               true	bool
		IsSelfBusy              false	bool
		IsSelfDirty		false	bool
		IsSelfValid		true	bool
		IsValid			true	bool

For the last 6 hours I have tried a number of permutations and I still can't get it to work.  Does anyone have any ideas on what I should do.  I'm really thinking of using reflection to change the isDirty flag back to false.  But I loath hacks.

**Comment by rockfordlhotka** [comment, marked as answer]
When is `IsDirty` (and `IsSelfDirty`?) set as true? At the bottom of your factory method, or after the data portal returns, or somewhere else?

**Reply by kcabral817** [reply]
The IsSelfDirty is false.  The object IsDirty once it gets back from the automapper (MapContainer.Map(info,obj) method.

**Reply by rockfordlhotka** [reply]
That is telling - a good clue.

`IsSelfDirty` represents the state of the immediate object and its properties, but not its child objects.

`IsDirty` represents the state of the object _graph_ including the current object and all its children.

So what I think I'm hearing is that `IsSelfDirty` is false, but `IsDirty` is true? In that case one of the child objects has been changed (or loaded incorrectly).

**Comment by kcabral817** [comment]
Awesome clue!  I got it (Irish jig).  Because I use .include(s) in my entity framework queries, I figured out that I'm responsible for checking the object graph to set the retrieved values as MarkOld().  When I found the code table that was loaded, and set the property, all is well!   Thank you so much for that clue :-)
